pkg/networkservice/chains/client/client.go

```diff
@@ -56,13 +56,13 @@ func NewClient(ctx context.Context, clientOpts ...Option) networkservice.Network
 				opts.refreshClient,
 				clienturl.NewClient(opts.clientURL),
 				clientconn.NewClient(opts.cc),
-			},
-			append(
-				opts.additionalFunctionality,
```

Comment:
Why do we need to change this?

Reply:
In case of URL change `dial.Client` does the following on Request:
1. Close old URL.
2. Request new URL.

If we have `sendfd.Client` in the chain before it does the following on Request:
1. insert `grpcfd.PerRPCCredentials` to the call options.

`grpcfd.PerRPCCredentials` does the following:
1. Store fds that needs to be sent.
2. On first gRPC interaction send them and remember connection as a `FDTransceiver`.
3. Cleanup all stored fds.

So it results in the following:
1. `dial` closes old URL - `PerRPCCredentials` sends all fds to the old URL.
2. `dial` requests new URL - `PerRPCCredentials` has nothing to send so it does nothing.

So here are 2 solutions that I can currently see:
1. Put `dial` client before `additionalFunctionality`.
2. Edit `PerRPCCredentials` not to capture `FDTransceiver` and not to cleanup stored fds on send.

We have discussed this with @edwarnicke and so it actually looks like both of these solutions have their pros and cons, but [1] is easier to implement.